internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -268,9 +268,9 @@ func (r *BundleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&rukpakv1alpha1.Bundle{}, builder.WithPredicates(
 			util.BundleProvisionerFilter(plain.ProvisionerID),
 		)).
-		// The default unpacker creates Pod's ownerref'd to its bundle, so
-		// we need to watch pods to ensure we reconcile events coming from these
-		// pods.
-		Watches(&crsource.Kind{Type: &corev1.Pod{}}, util.MapOwneeToOwnerProvisionerHandler(context.Background(), mgr.GetClient(), l, plain.ProvisionerID, &rukpakv1alpha1.Bundle{})).
+		// The default image source unpacker creates Jobs ownerref'd to its bundle, so
+		// we need to watch jobs to ensure we reconcile events coming from these
+		// jobs.
+		Watches(&crsource.Kind{Type: &batchv1.Job{}}, util.MapOwneeToOwnerProvisionerHandler(context.Background(), mgr.GetClient(), l, plain.ProvisionerID, &rukpakv1alpha1.Bundle{})).
```

Comment:
Is it enough just to watch the Job?  Do we care about the Pod deleted manually?

Reply:
Yep, good catch. I had also found another corner case where watching just jobs is not enough granularity for bubbling up specific events from pods (e.g. the e2e is failing the image pull backoff test)